tools/bias_lang_migration.sh

```diff
@@ -0,0 +1,604 @@
+
+	for pod in ${PODS}; do
+		echo "Adding pod ${pod} to ${MANAGER}"
+		kubectl -n ${NS} exec -it ${pod} -- /bin/bash -c "${command}"
+		if [[ "$?" -ne 0 ]]; then
+			echo "Failed to add peer to new Manager"
+		fi
+	done
+}
+
+add_node_affinity() {
+	FILE=$1
+	cp ${FILE} ${TMP_FOLDER}/temp.node.info.${NS}.${TT}
+	cat ${TMP_FOLDER}/temp.node.info.${NS}.${TT} | jq '.spec += {
+          "affinity": {
+              "nodeAffinity": {
+                    "requiredDuringSchedulingIgnoredDuringExecution": {
+                          "nodeSelectorTerms": [
+                    {
+                "matchExpressions": [
+                  {
+                    "key": "biaslangmasternode",
+                    "operator": "In",
+                    "values": [
+                        "yes"
+                    ]
+                  }
+                ]
+              }
+          ]}}}}' >${FILE}
+}
+
+reset_manager_CR() {
+	FILE=$1
+	name=$(cat ${FILE} | jq '.metadata.name' -r)
+	type=$(cat ${FILE} | jq '.kind' -r)
+
+	case $type in
+
+	ClusterManager)
+		target="cluster-manager"
+		;;
+
+	LicenseManager)
+		target="license-manager"
+		;;
+
+	*)
+		echo -n "invalid reset request for CR type ${type}"
+		;;
+	esac
+
+	kubectl -n ${NS} delete pod "splunk-${name}-${target}-0"
+	is_pod_ready "splunk-${name}-${target}-0"
+}
+
+# Template to create Rsync Jobs from Master to Manager
+create_job() {
+	json='{
+          "apiVersion": "batch/v1",
+          "kind": "Job",
+          "metadata": {
+            "name": "rsync-$3-pvc-$4",
+            "finalizers": [
+              "foregroundDeletion"
+            ]
+          },
+          "spec": {
+            "template": {
+              "spec": {
+                "volumes": [
+                  {
+                    "name": "src-dir",
+                    "persistentVolumeClaim": {
+                      "claimName": "$1"
+                    }
+                  },
+                  {
+                    "name": "dest-dir",
+                    "persistentVolumeClaim": {
+                      "claimName": "$2"
+                    }
+                  }
+                ],
+                "containers": [
+                  {
+                    "name": "alpine",
+                    "image": "alpine:latest",
+                    "command": [
+                      "sh",
+                      "-c",
+                      "apk add --update rsync && rsync -azPv /src-dir/ /dest-dir && ls -la /dest-dir"
+                    ],
+                    "volumeMounts": [
+                      {
+                        "mountPath": "/src-dir",
+                        "name": "src-dir"
+                      },
+                      {
+                        "mountPath": "/dest-dir",
+                        "name": "dest-dir"
+                      }
+                    ],
+                    "resources": {
```

Comment:
how do we know the resource sizing is same as original one?

Reply:
As discussed in the last meeting, we set this hardcoded value as the minimum needed to perform the copy.